chinglee-iot/FreeRTOS-Kernel
# Merge prvCheckTasksWaitingTermination from SMP

Merge prvCheckTasksWaitingTermination from SMP branch

Description
-----------


Test Steps
-----------
* WIN32-MSVC test application
* Raspberry pi pico test application
* Raspberry pi pico smp build

Related Issue
-----------


Performance Comparison Test
-----------
* Extra check for SMP.
* The performance has no effect but the code size increase a little bit due to extra check.

|                                                  | smp-dev-prvCheckTasksWaitingTermination ( -O0 ) |
| ------------------------------------------------ | --------------------------------------- |
| PerfTest\_vTaskStartScheduler                    | 0                                       |
| PerfTest\_vTaskDelete\_DeleteCurrentTask         | 0                                       |
| PerfTest\_vTaskDelete\_DeleteOtherTask           | 0                                       |
| PerfTest\_vTaskPrioritySet\_SetOtherTaskHigh     | 0                                       |
| PerfTest\_vTaskPrioritySet\_SetOtherTaskLow      | 0                                       |
| PerfTest\_vTaskPrioritySet\_SetCurrentTaskHigh   | \-1                                     |
| PerfTest\_vTaskPrioritySet\_SetCurrentTaskLow    | 0                                       |
| PerfTest\_vTaskPrioritySet\_TaskSwitch           | 0                                       |
| PerfTest\_vTaskSuspend\_SuspendCurrent           | 0                                       |
| PerfTest\_vTaskSuspend\_SuspendOther             | 0                                       |
| PerfTest\_vTaskSuspend\_TaskSwitch               | 0                                       |
| PerfTest\_vTaskResume\_ResumeLowTask             | 0                                       |
| PerfTest\_vTaskResume\_ResumeHighTask            | 0                                       |
| PerfTest\_xTaskResumeFromISR\_ResumeLowTask      | 0                                       |
| PerfTest\_xTaskResumeFromISR\_ResumeHighTask     | 0                                       |
| PerfTest\_xTaskResumeFromISR\_TaskSwitch         | 0                                       |
| PerfTest\_xTaskAbortDelay\_HighPriorityTask      | 0                                       |
| PerfTest\_xTaskAbortDelay\_LowPriorityTask       | \-1                                     |
| PerfTest\_xTaskAbortDelay\_TaskSwitch            | 0                                       |
| PerfTest\_xTaskIncrementTick\_SchedulerSuspended | 0                                       |
| PerfTest\_xTaskIncrementTick\_SchedulerRunning   | 0                                       |
| PerfTest\_xTaskIncrementTick\_TaskSwitch         | 0                                       |
| PerfTest\_vTaskSwtichContext\_SwitchToCurrent    | 0                                       |
| PerfTest\_vTaskSwtichContext\_SwitchToOther      | 0                                       |
| PerfTest\_vTaskSwtichContext\_TaskSwitch         | 0                                       |
| PerfTest\_xQueueSend\_UnblockHighTask            | 0                                       |
| PerfTest\_xQueueSend\_UnblockLowTask             | 0                                       |
| PerfTest\_xQueueSend\_TaskSwitch                 | \-1                                     |
| PerfTest\_xEventGroupsSetBits\_unblockHighTask   | 0                                       |
| PerfTest\_xEventGroupsSetBits\_unblockLowTask    | 0                                       |
| PerfTest\_xEventGroupSetBits\_TaskSwitch         | 0                                       |
| PerfTest\_xTaskNotifyGive\_UnblockHighTask       | 0                                       |
| PerfTest\_xTaskNotifyGive\_UnblockLowTask        | \-4                                     |
| PerfTest\_xTaskNotifyGive\_TaskSwitch            | 0                                       |
| PerfTest\_xTaskNotifyFromISR\_NotifyHighTask     | 0                                       |
| PerfTest\_xTaskNotifyFromISR\_NotifyLowTask      | 0                                       |
| PerfTest\_vTaskNotifyGiveFromISR\_NotifyHighTask | 0                                       |
| PerfTest\_vTaskNotifyGiveFromISR\_NotifyLowTask  | 0                                       |
| xAvailableHeapSpaceInBytes                       | 0                                       |
| xNumberOfSuccessfulAllocations                   | 0                                       |
| xNumberOfSuccessfulFrees                         | 0                                       |
| xMinimumEverFreeBytesRemaining                   | 0                                       |
| text                                             | 48                                      |
| data                                             | 0                                       |
| bss                                              | 0                                       |


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
